src/kakarot/instructions/system_operations.cairo

```diff
@@ -511,6 +511,7 @@ namespace CallHelper {
         // TODO: would break the whole computation, so if it does not, it's TRUE
         let success = Uint256(low=1, high=0);
         let ctx = ExecutionContext.update_sub_context(ctx.calling_context, ctx);
+        let ctx = ExecutionContext.increment_gas_used(ctx, ctx.sub_context.gas_used);
```

Comment:
is this a duplicate of line 619?

Reply:
yes, I noticed that the sub_context gas_usage was not reported in finalize_call. We could however actually move this to the `instructions::run` https://github.com/sayajin-labs/kakarot/blob/ba15ba3726ef8f67edd50b156e6034512563ec6a/src/kakarot/instructions.cairo#L625

Reply:
not so good to put `let ctx = ExecutionContext.increment_gas_used(ctx, ctx.sub_context.gas_used);` in `instructions` because we'd need to put one for each early return.